Origin: http://libvirt.org/git/?p=libvirt.git;a=commitdiff;h=3cdb3b4e#patch1
 http://libvirt.org/git/?p=libvirt.git;a=commitdiff;h=d59e1033#patch1
Description: In vshSnapshotListCollect() vshCalloc was called with swapped
 nmemb and size argument. This caused division by zero in xalloc_oversized as
 the macro doesn't expect size to be zero. Also, fill the domain argument that
 is sent to the remote side.
Bug-Ubuntu: https://launchpad.net/bugs/1039678

Index: libvirt-0.9.13/tools/virsh.c
===================================================================
--- libvirt-0.9.13.orig/tools/virsh.c	2012-08-21 13:54:12.000000000 -0500
+++ libvirt-0.9.13/tools/virsh.c	2012-08-21 13:56:20.000000000 -0500
@@ -17202,8 +17202,8 @@
     if (count >= 0) {
         /* When mixing --from and --tree, we also want a copy of from
          * in the list, but with no parent for that one entry.  */
-        snaplist->snaps = vshCalloc(ctl, sizeof(*snaplist->snaps),
-                                    count + (tree && from));
+        snaplist->snaps = vshCalloc(ctl, count + (tree && from),
+                                    sizeof(*snaplist->snaps));
         snaplist->nsnaps = count;
         for (i = 0; i < count; i++)
             snaplist->snaps[i].snap = snaps[i];
Index: libvirt-0.9.13/src/remote/remote_driver.c
===================================================================
--- libvirt-0.9.13.orig/src/remote/remote_driver.c	2012-06-25 02:06:18.000000000 -0500
+++ libvirt-0.9.13/src/remote/remote_driver.c	2012-08-21 13:57:20.000000000 -0500
@@ -4888,6 +4888,7 @@
 
     remoteDriverLock(priv);
 
+    make_nonnull_domain(&args.dom, dom);
     args.need_results = !!snapshots;
     args.flags = flags;
 
